docs/api/checkouts.rst

```diff
@@ -238,185 +563,307 @@ Response will contain an updated order information.
-                },
-                "state": "checkout",
-                "updated_at": "2014-12-15T14:30:41+0000"
+                ]
             }
-        ],
-        "shipping_address": {
-        },
-        "state": "cart",
-        "total": 1504750,
-        "updated_at": "2014-12-15T14:30:41+0000",
-        "user": {
-        }
+        ]
     }
 
+Response contains the proposed shipments and for each, it also has a list of shipping methods available with calculated price.
 
-Payment step
-------------
+.. warning::
 
-When we are done with shipping choices and we know the final price of an order, we can select a payment method.
+    Because of custom calculation logic for this endpoint the regular rules of overriding does not apply for this endpoint. In order to achieve different message one has to provide custom controller and build the message by his own. Exemplary implementation can be found `here`__
+
+Next step is updating the order with the types of shipping method that one has selected. A PUT request has to be send for each available shipment,
 
-To obtain a list of available payment methods for this order, simply call a GET request again:
+Definition
+^^^^^^^^^^
 
 .. code-block:: text
 
-    GET /api/v1/checkouts/44
+    PUT /api/v1/checkouts/select-shipping/{id}
+
++------------------------+----------------+--------------------------------------+
+| Parameter              | Parameter type | Description                          |
++========================+================+======================================+
+| Authorization          | header         | Token received during authentication |
++------------------------+----------------+--------------------------------------+
+| id                     | url attribute  | Id of the requested cart             |
++------------------------+----------------+--------------------------------------+
+| shipments[X]['method'] | request        | Code of chosen shipping method       |
++------------------------+----------------+--------------------------------------+
+
+Where X is a number of shipment in returned array.
+
+Example
+^^^^^^^
+
+To choose a `DHL Express` method for our shipment (the cheapest one), one could use the following code:
+
+.. code-block:: bash
+
+    $ curl http://127.0.0.1:8000/api/v1/checkouts/select-shipping/21 \
+        -H "Authorization: Bearer SampleToken" \
+        -H "Content-Type: application/json" \
+        -X PUT \
+        --data '
+            {
+                "shipments": [
+                    {
+                        "method": "dhl_express"
+                    }
+                ]
+            }
+        '
+
+Exemplary Response
+^^^^^^^^^^^^^^^^^^
 
 .. code-block:: text
 
-    STATUS: 200 OK
+    STATUS: 204 No Content
 
-.. code-block:: json
+Now you can check the state of order asking for checkout summary just like before:
+
+Example
+^^^^^^^
+
+To check the checkout process state for cart with `id = 21`, we need to execute this command:
```

Comment:
Again duplication.

Try doing sth like

```
While checking for the checkout process state of the cart with `id = 21`, you will get the following response:
```
 Without the whole `Example` section.
Instead of copying and pasting the same command so many times.